### PR TITLE
refactor/heatmap types

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Refactoring, cleaning, optimizing and typesafing heatmap series.

### CODE REVIEW

File: ts/Series/Heatmap/HeatmapPoint.ts [hunk 1]
```diff
@@ -85,16 +85,12 @@ class HeatmapPoint extends ScatterPoint {
             delete this.color;
         }
 
-        const point: HeatmapPoint = super.applyOptions.call(
-            this,
-            options,
-            x
-        ) as any;
+        super.applyOptions.call(this, options, x);
```
Comment (bre1470) [inline ts/Series/Heatmap/HeatmapPoint.ts:88]: Good catch. Actually `super.applyOptions(options, x);` is sufficient.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Looks nice! Please see comments for optimizations.

File: ts/Series/ColorMapComposition.ts [hunk 2]
```diff
@@ -167,14 +175,14 @@ namespace ColorMapComposition {
     function seriesColorAttribs(
         this: SeriesComposition,
         point: PointComposition
-    ): SVGAttributes {
-        const ret: SVGAttributes = {};
+    ): ColorAttribsType {
```
[the remark this answers is not in the document]
Comment (TorsteinHonsi) [reply ts/Series/ColorMapComposition.ts:178]: It is by design. ColorAttribsType is a subset of SVGAttributes and CSSObject.

**Review by MarkusBarstad — APPROVED** [review]
A lot more readable!

**Review by bre1470 — APPROVED** [review]

**Pull request merged**